ddtrace/contrib/grpc/server_interceptor.py

```diff
@@ -0,0 +1,141 @@
+import grpc
+from ddtrace.vendor import wrapt
+
+from ddtrace import config
+from ddtrace.ext import errors
+from ddtrace.compat import to_unicode
+
+from ...constants import ANALYTICS_SAMPLE_RATE_KEY
+from ...propagation.http import HTTPPropagator
+from . import constants
+from .utils import parse_method_path
+
+
+def create_server_interceptor(pin):
+    def interceptor_function(continuation, handler_call_details):
+        if not pin.enabled:
+            return continuation(handler_call_details)
+
+        rpc_method_handler = continuation(handler_call_details)
+        return _TracedRpcMethodHandler(pin, handler_call_details, rpc_method_handler)
+
+    return _ServerInterceptor(interceptor_function)
+
+
+def _handle_server_exception(server_context, span):
+    if server_context is not None and \
+       hasattr(server_context, '_state') and \
+       server_context._state is not None:
+        code = to_unicode(server_context._state.code)
+        details = to_unicode(server_context._state.details)
+        span.error = 1
+        span.set_tag(errors.ERROR_MSG, details)
+        span.set_tag(errors.ERROR_TYPE, code)
+
+
+def _wrap_response_iterator(response_iterator, server_context, span):
+    try:
+        for response in response_iterator:
```

Comment:
can we reuse the wrap iterator thing you have for the client response?

Reply:
The server exception has to be handled differently since we have to access the server context and don't yet have a `grpc.RpcError` exception to handle (see `_handle_server_exception` above).